### PR TITLE
Fix VPN configuration removal to stop the tunnel

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/1207603085593419/1207832283330964/f

macOS PR: https://github.com/duckduckgo/macos-browser/pull/2991
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/900

## Description:

Fix the VPN configuration removal handling in the extension to stop the tunnel.

## Testing:

There's an option in the debug menu to remove the VPN configuration.

1. Start the VPN.
2. Test that the tunnel works.
3. See the monitors are running in Console.app.
4. Debug menu > VPN > Reset > Remove VPN Configuration
5. Check that the VPN is stopped
6. Check in Console.app that the monitors are stopped
7. Start the VPN again and ensure it works fine

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: submodules/privacy-reference-tests
Comment (samsymons) [inline submodules/privacy-reference-tests:1]: Is it intentional to include the Privacy Reference Tests bump?

**Review by samsymons — APPROVED** [review]
LGTM with one comment about an unintentional change.

File: submodules/privacy-reference-tests
Comment (diegoreymendez) [reply submodules/privacy-reference-tests:1]: Yes, completely unintentional, thanks for spotting it!

File: DuckDuckGo.xcodeproj/xcshareddata/xcschemes/DuckDuckGo.xcscheme [hunk 1]
```diff
@@ -64,6 +64,9 @@
                <Test
                   Identifier = "AutoconsentMessageProtocolTests/testEval()">
                </Test>
+               <Test
+                  Identifier = "SyncSettingsViewControllerErrorTests/test_WhenAccountRemoved_ErrorHandlerSyncDidTurnOffCalled()">
+               </Test>
```
Comment (diegoreymendez) [inline DuckDuckGo.xcodeproj/xcshareddata/xcschemes/DuckDuckGo.xcscheme:69]: Disabling flaky test.

**Pull request merged**